File: typescript-client-sdk/src/sdk/models/operations/getcompaniescompanyidreportsagedcreditor.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetCompaniesCompanyIdReportsAgedCreditorPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=companyId" })
  companyId: string;
}


export class GetCompaniesCompanyIdReportsAgedCreditorQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=numberOfPeriods" })
  numberOfPeriods?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=periodLengthDays" })
  periodLengthDays?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=reportDate" })
  reportDate?: Date;
}


export class GetCompaniesCompanyIdReportsAgedCreditorSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeApiKeyAuth;
}


export class GetCompaniesCompanyIdReportsAgedCreditorRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetCompaniesCompanyIdReportsAgedCreditorPathParams;

  @SpeakeasyMetadata()
  queryParams: GetCompaniesCompanyIdReportsAgedCreditorQueryParams;

  @SpeakeasyMetadata()
  security: GetCompaniesCompanyIdReportsAgedCreditorSecurity;
}


export class GetCompaniesCompanyIdReportsAgedCreditorResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  codatDataContractsDatasetsAgedCreditorOutstandingICollectionReport?: shared.CodatDataContractsDatasetsAgedCreditorOutstandingICollectionReport;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: go-client-sdk/pkg/models/operations/getcompaniescompanyidreportsagedcreditor.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/v2/pkg/models/shared"
	"time"
)

type GetCompaniesCompanyIDReportsAgedCreditorPathParams struct {
	CompanyID string `pathParam:"style=simple,explode=false,name=companyId"`
}

type GetCompaniesCompanyIDReportsAgedCreditorQueryParams struct {
	NumberOfPeriods  *int       `queryParam:"style=form,explode=true,name=numberOfPeriods"`
	PeriodLengthDays *int       `queryParam:"style=form,explode=true,name=periodLengthDays"`
	ReportDate       *time.Time `queryParam:"style=form,explode=true,name=reportDate"`
}

type GetCompaniesCompanyIDReportsAgedCreditorSecurity struct {
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
}

type GetCompaniesCompanyIDReportsAgedCreditorRequest struct {
	PathParams  GetCompaniesCompanyIDReportsAgedCreditorPathParams
	QueryParams GetCompaniesCompanyIDReportsAgedCreditorQueryParams
	Security    GetCompaniesCompanyIDReportsAgedCreditorSecurity
}

type GetCompaniesCompanyIDReportsAgedCreditorResponse struct {
	CodatDataContractsDatasetsAgedCreditorOutstandingICollectionReport *shared.CodatDataContractsDatasetsAgedCreditorOutstandingICollectionReport
	ContentType                                                        string
	StatusCode                                                         int
}
